src/seamless_communication/cli/m4t/classification_head/train.py

```diff
@@ -0,0 +1,295 @@
+        ),
+    )
+    parser.add_argument(
+        "--device",
+        type=str,
+        default="cuda",
+        help=("Device to fine-tune on. See `torch.device`."),
+    )
+    parser.add_argument(
+        "--num_languages", 
+        type=int, 
+        default=2, 
+        help="The number of classes"
+    )
+    parser.add_argument(
+        "--num_layers", 
+        type=int, 
+        default=2, 
+        help="The number of layers in the classification head"
+    )
+    return parser
+
+
+def train(head: torch.nn.Module,
+            frozen_model: UnitYModel,
+            dataloader: dataloader.UnitYDataLoader,
+            params: ClassificationHeadTrainParams,
+            label_weights: torch.Tensor = None):
+    
+    logger.info("Start Training Language Head")
+    dataloader = dataloader.get_dataloader()
+    frozen_model.train()
+    
+    grad_scaler = torch.cuda.amp.GradScaler()
+    optimizer = AdamW(
+        params=frozen_model.parameters(),
+        lr=params.learning_rate,
+        betas=(0.9, 0.98),
+        eps=1e-08,
+        maximize=False,
+        weight_decay=0.0,
+        fused=(params.device.type == "cuda"))
+    lr_scheduler = MyleLR(
+        optimizer=optimizer,
+        num_warmup_steps=params.warmup_steps,
+        start_lr=1e-9)
+
+    losslog = list()
+    try:
+        for epoch in range(params.max_epochs):
+            logger.info(f"Epoch {epoch}")
+            epoch_loss = 0.0
+            for batch in tqdm(dataloader, desc="Training Steps"):
+                # Run batch through train step
+                optimizer.zero_grad()
+                with torch.autocast(device_type=params.device.type, dtype=params.float_dtype):
+                    vector, _ = frozen_model.encode(batch)
+                
+                _y = head(vector)
+                
+                loss = torch.nn.functional.cross_entropy(batch, _y, weight=label_weights)
+                if loss.isnan().any().item():
+                    logger.error(batch.speech_to_text)
+                    raise RuntimeError("Train loss is NaN! Something is wrong in the model!")
+                losslog.append(loss.item())
+                epoch_loss += loss.item()
+                
+                grad_scaler.scale(loss).backward()
+                grad_scaler.step(optimizer)
+                grad_scaler.update()
+                lr_scheduler.step()
+                
+                assert batch.speech_to_text.src_tokens is not None
+            logger.info(f"Epoch {epoch} Loss: {epoch_loss / len(dataloader)}")
+
+    except KeyboardInterrupt:
+        logger.info("Interrupted by user. Saving model state...")
+        torch.save({
+            'model_state_dict': head.state_dict(),
+            'optimizer_state_dict': optimizer.state_dict(),
+            'loss': loss,
+        }, "interrupted_model.pth")
+        logger.info("Model state saved. Exiting...")
+        exit()
+
+    logger.info(f"Final Loss: {sum(losslog) / len(losslog)}")
```

Comment:
This would give the average loss, which doesn't really make that much sense anyway